File: Professional/ActiveReportsWithMVC5WebAPI/C#/ActiveReportsWithMVC5/Controllers/CustomersController.cs
using ActiveReportsWithMVC5.Models;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
namespace ActiveReportsWithMVC5.Controllers
{
	public class CustomersController : ApiController
	{
		//Return customers
		public IEnumerable<Customer> Get()
		{
			return CustomerContext.GetCustomers().AsEnumerable();
		}
		//Return customer based on CustomerID
		public Customer Get(int id)
		{
			Customer customer = CustomerContext.GetCustomersDetail(id);			   
			if(customer==null)
			{
				throw new HttpResponseException(Request.CreateResponse(HttpStatusCode.NotFound));
			}
			return customer;
		}	   
	}
}

File: Professional/ActiveReportsWithMVC5WebAPI/VB.NET/ActiveReportsWithMVC5/Controllers/CustomersController.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Net
Imports System.Net.Http
Imports System.Web.Http
Public Class CustomersController
	Inherits ApiController
	' Return customers
	Public Function GetCustomer() As IEnumerable(Of Customer)
		Return CustomerContext.GetCustomers().AsEnumerable()
	End Function
	' Return customer based on CustomerID
	Public Function GetCustomer(id As Integer) As Customer
		Dim customer As Customer = CustomerContext.GetCustomersDetail(id)
		If customer Is Nothing Then
			Throw New HttpResponseException(Request.CreateResponse(HttpStatusCode.NotFound))
		End If
		Return customer
	End Function
End Class
